### PR TITLE
Local storage

### PR DESCRIPTION
**Pull request by cicdw**
**Thanks for contributing to Prefect!**

Please describe your work and make sure your PR:

- [x] adds new tests (if appropriate)
- [x] updates `CHANGELOG.md` (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

Note that your PR will not be reviewed unless all three boxes are checked.

## What does this PR change?
Adds a new `Storage` class for storing flows in your local filesystem.  Finishes out most of the "natural" storage options for now.

## Why is this PR important?
Adds a useful permanent storage option for open source users.

### CODE REVIEW

File: src/prefect/environments/storage/local.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+
+class LocalStorage(Storage):
+    """
+    Local Storage class.  This class represents the Storage
+    interface for Flows stored as bytes in the local filesystem.
+
+    Args:
+        - directory (str, optional): the directory the flows will be stored in;
+            defaults to `~/.prefect`
```
Comment (joshmeek) [inline src/prefect/environments/storage/local.py]: What are your thoughts on a `~/.prefect/flows` subdirectory as default?

Comment (joshmeek) [reply src/prefect/environments/storage/local.py]: Well actually then that would require creation of the directory

Comment (jlowin) [reply src/prefect/environments/storage/local.py]: Actually I like this idea, because it would give users a "safe" directory they could delete in one shot

Comment (cicdw) [reply src/prefect/environments/storage/local.py]: Yea I can do that; I can create the directory, no big deal.

Comment (cicdw) [reply src/prefect/environments/storage/local.py]: Updated -- good call!

File: src/prefect/environments/storage/local.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        - directory (str, optional): the directory the flows will be stored in;
+            defaults to `~/.prefect/flows`.  If it doesn't already exist, it will be
+            created for you.
+    """
+
+    def __init__(self, directory: str = "~/.prefect/flows") -> None:
+        self.flows = dict()  # type: Dict[str, str]
+        abs_directory = os.path.abspath(os.path.expanduser(directory))
+        if not os.path.exists(abs_directory):
+            os.makedirs(abs_directory)
```
Comment (jlowin) [inline src/prefect/environments/storage/local.py]: Doesn't matter, but this also accepts `exists_ok=True`

Comment (cicdw) [reply src/prefect/environments/storage/local.py]: Ah yes, forgot about that - happy to change if you want!

Comment (jlowin) [reply src/prefect/environments/storage/local.py]: Nah, this is totally fine, was mentioning in case you didn't know

**Pull request merged**